test/services/data_model_engine.js

```diff
@@ -1160,6 +1160,36 @@ describe('Data Model Engine', () => {
       expect(result.failed.bundle3.message).to.equal('An error');
       expect(result.failed.bundle3.stack).to.exist;
     });
+
+    it('retries uploads when rate limited', async () => {
```

Comment:
This test takes 3 seconds to complete as it actually calls the real bluebird delay function internally.